src/lustre/lustreDesugarIfBlocks.ml

```diff
@@ -0,0 +1,344 @@
+(* This file is part of the Kind 2 model checker.
+
+   Copyright (c) 2022 by the Board of Trustees of the University of Iowa
+
+   Licensed under the Apache License, Version 2.0 (the "License"); you
+   may not use this file except in compliance with the License.  You
+   may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0 
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+   implied. See the License for the specific language governing
+   permissions and limitations under the License. 
+ *)
+
+ (** 
+   Code for desugaring imperative-style if blocks to functional ITEs.
+
+   The code has a few steps.
+    1. Remove multiple assignment from if blocks using temp variables.
+    2. Parse the if block and create a map of trees, one for each variable
+    3. Fill in oracles in the trees for missing equations.
+    4. Remove redundancy from the trees.
+    5. Convert the trees to ITE expressions. 
+
+   @author Rob Lorch
+ *)
+
+module R = Res
+module A = LustreAst
+module AH = LustreAstHelpers
+module GI = GeneratedIdentifiers
+module Chk = LustreTypeChecker
+module Ctx = TypeCheckerContext
+
+
+type error_kind = Unknown of string
+  | MisplacedNodeItemError of A.node_item
+
+let error_message error = match error with
+  | Unknown s -> s
+  | MisplacedNodeItemError ni -> (match ni with
+    | Body (Assert _) -> "Asserts are not allowed inside frame blocks."
+    | FrameBlock _ -> "Frame blocks are not allowed inside other frame blocks."
+    | AnnotMain _ -> "Main annotations are not allowed inside frame blocks."
+    | AnnotProperty _ -> "Property annotations are not allowed inside frame blocks."
```

Comment:
All these error messages should say: `[...] inside if blocks` instead of `[...] inside frame blocks`